cli/create.go

```diff
@@ -95,20 +97,11 @@ func cmdCreateContainer(context *cli.Context, attach bool) error {
 	} else {
 		for _, ns := range spec.Linux.Namespaces {
 			if ns.Path != "" {
-				if strings.Contains(ns.Path, "/") {
-					return fmt.Errorf("Runv doesn't support path to namespace file, it supports containers name as shared namespaces only")
-				}
 				if ns.Type == "mount" {
 					return fmt.Errorf("Runv doesn't support containers with shared mount namespace, use `runv exec` instead")
 				}
-				sharedContainer = ns.Path
-				_, err = os.Stat(filepath.Join(root, sharedContainer, stateJSON))
-				if err != nil {
-					return fmt.Errorf("The container %q is not existing or not ready", sharedContainer)
-				}
-				_, err = os.Stat(filepath.Join(root, sharedContainer, "namespace"))
-				if err != nil {
-					return fmt.Errorf("The container %q is not ready", sharedContainer)
+				if sharedContainer, err = findSharedContainer(context.GlobalString("root"), ns.Path); err != nil {
+					return fmt.Errorf("failed to find shared container: %v", err)
```

Comment:
Can you keep 
```
-				_, err = os.Stat(filepath.Join(root, sharedContainer, stateJSON))
 -				if err != nil {
 -					return fmt.Errorf("The container %q is not existing or not ready", sharedContainer)
 -				}
 -				_, err = os.Stat(filepath.Join(root, sharedContainer, "namespace"))
 -				if err != nil {
 -					return fmt.Errorf("The container %q is not ready", sharedContainer)
```

This piece of code is used for validation, I think it's still useful.

Reply:
it's not useful.
```
 	if sharedContainer != "" {
 		scState, err = loadStateFile(root, sharedContainer)
 		if err != nil {
 			return err
 		}
 		vm, lockFile, err = getSandbox(filepath.Join(context.GlobalString("root"), sharedContainer, "sandbox"))
 		if err != nil {
 			return err
 		}
 	}
```
```
func loadStateFile(root, container string) (*State, error) {
	stateFile := filepath.Join(root, container, stateJSON)
	file, err := os.Open(stateFile)
	if err != nil {
		return nil, err
	}
	defer file.Close()

	var state State
	if err = json.NewDecoder(file).Decode(&state); err != nil {
		return nil, fmt.Errorf("Decode state file %s error: %s", stateFile, err.Error())
	}
	return &state, nil
}
```
loadStateFile will confirm the state file exist.
namespace file is not used.

Reply:
Looks like it's OK then.

ping @laijs